src/validators/validateCreateNotificationParams.ts

```diff
@@ -0,0 +1,12 @@
+import Joi from 'joi';
+import type { Notification } from '../services/notification.service';
+
+export const validateCreateNotificationParams = (
+  data: Notification,
+): Joi.ValidationResult => Joi.object().keys({
+  notificationType: Joi.string().required(),
+  toUserId: Joi.number().min(1).required(),
```

Comment:
I turns out that, as written, Joi will allow floating point numbers, which will then cause PostgreSQL to throw an error:

```
$ curl -v -X POST -H "Content-Type: application/json" --data '{"toUserId":4.3,"notificationType":"not an integer"}' http://localhost:3000/api/notifications | jq .
(trimmed)
< HTTP/1.1 500 Internal Server Error
(trimmed)
{
  "error": {
    "name": "TinyPgError",
    "queryContext": {
      "id": "c8926922-8000-4c4e-9f7c-2e329e63af93",
      "sql": "INSERT \nINTO notifications (to_user_id, notification_type) \nVALUES ($1, $2)\nRETURNING notification_id\n",
      "start": 1615243221034,
      "name": "insert-notification",
      "params": {
        "to_user_id": 4.3,
        "notification_type": "not an integer"
      },
      "submit": 1615243221039,
      "wait_duration": 5,
      "active_duration": 30,
      "end": 1615243221069,
      "duration": 35,
      "error": {
        "length": 104,
        "name": "error",
        "severity": "ERROR",
        "code": "22P02",
        "file": "numutils.c",
        "line": "323",
        "routine": "pg_strtoint32"
      },
      "data": null
    }
  }
}
```

It looks like Joi has an [integer requirement](https://joi.dev/api/?v=17.4.0#numberinteger), which is pretty straightforward to add:

```suggestion
  toUserId: Joi.number().integer().min(1).required(),
```

Reply:
Oh thank you for testing that. Yes will make changes.